concordium-smart-contract-testing/src/types.rs

```diff
@@ -35,22 +37,52 @@ pub(crate) struct ChainParameters {
     pub(crate) euro_per_energy: ExchangeRate,
 }
 
+/// The connection and runtime needed for communicating with an external node.
+#[derive(Debug)]
+pub(crate) struct ExternalNodeConnection {
+    /// An instantiated v2 Client from the Rust SDK. Used for communicating with
+    /// a node.
+    pub(crate) client:       concordium_rust_sdk::v2::Client,
+    /// A Tokio runtime used to execute the async methods of the `client`.
+    pub(crate) runtime:      tokio::runtime::Runtime,
+    /// The sticky block hash to use for queries without specific blocks.
+    ///
+    /// The hash is "sticky" in the sense that it is set on the first query and
+    /// will be used for subsequent queries that do *not* specify block hash.
+    pub(crate) sticky_block: Option<BlockHash>,
+}
+
 /// Represents the blockchain and supports a number of operations, including
 /// creating accounts, deploying modules, initializing contract, updating
 /// contracts and invoking contracts.
 #[derive(Debug)]
 pub struct Chain {
-    pub(crate) parameters:          ChainParameters,
+    pub(crate) parameters:               ChainParameters,
     /// Accounts and info about them.
     /// This uses [`AccountAddressEq`] to ensure that account aliases are seen
     /// as one account.
-    pub(crate) accounts:            BTreeMap<AccountAddressEq, Account>,
+    pub(crate) accounts:                 BTreeMap<AccountAddressEq, Account>,
     /// Smart contract modules.
-    pub(crate) modules:             BTreeMap<ModuleReference, ContractModule>,
+    pub(crate) modules:                  BTreeMap<ModuleReference, ContractModule>,
     /// Smart contract instances.
-    pub(crate) contracts:           BTreeMap<ContractAddress, Contract>,
+    pub(crate) contracts:                BTreeMap<ContractAddress, Contract>,
     /// Next contract index to use when creating a new instance.
-    pub(crate) next_contract_index: u64,
+    pub(crate) next_contract_index:      u64,
+    /// An optional connection to an external node.
+    pub(crate) external_node_connection: Option<ExternalNodeConnection>,
+}
+
+/// A builder for the [`Chain`].
```

Comment:
Can we add a pointer on how to get a chain out of it, a link to a `.build` function.

Reply:
Yes.